lib/src/pubspec_config/pubspec_config.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:yaml/yaml.dart';
+
+part 'pubspec_config.freezed.dart';
+
+@freezed
+abstract class PubspecConfig implements _$PubspecConfig {
```

Comment:
`LokaliseConfig`? In factory method we have information that this is from pubspec. In my opinion this is implementation detail at this level.